drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (C) 2017 George Psimenos (gp7g14@soton.ac.uk)
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_sht2x
+ * @{
+ *
+ * @file
+ * @brief       Driver for the Sensirion SHT2x sensor series
+ *
+ * @author      George Psimenos <gp7g14@soton.ac.uk>
+ *
+ * @}
+ */
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+
+#include "assert.h"
+#include "periph/i2c.h"
+#include "xtimer.h"
+#include "sht2x.h"
+#include "sht2x_regs.h"
+
+#define ENABLE_DEBUG        (0)
+#include "debug.h"
+
+#define I2C_SPEED           I2C_SPEED_FAST
+#define MAX_RETRIES         20
+
+#define BUS                 (dev->params.i2c)
+#define ADDR                (dev->params.addr)
+
+int sht2x_init(sht2x_t *dev, const sht2x_params_t *params)
+{
+    uint8_t result;
```

Comment:
You can get rid of this variable and directly use the result of i2c functions in the `if` statements below.